posthog/api/feature_flag.py

```diff
@@ -224,7 +224,7 @@ def properties_all_match(predicate):
                         for cohort in [initial_cohort, *dependent_cohorts]:
                             if [prop for prop in cohort.properties.flat if prop.type == "behavioral"]:
                                 raise serializers.ValidationError(
-                                    detail=f"Cohort '{cohort.name}' with behavioral filters cannot be used in feature flags.",
+                                    detail=f"Cohort '{cohort.name}' with behavioral filters cannot be used in feature flags or surveys.",
```

Comment:
Why shouldn't we have this message on the backend?

Reply:
It pops up as an error toast in the frontend anyway lol

Reply:
<img width="448" alt="Screenshot 2024-01-24 at 2 28 50 PM" src="https://github.com/PostHog/posthog/assets/25164963/69834cb2-008d-4c22-bf80-b9256f662bed">

Reply:
It usually helps also to give a reason if you're making suggestions for something 😅

Reply:
Ah, fair my bad 😅 - because it now shows up on feature flags too, saying 'surveys', and via API too, while the surveys bit really only matters on surveys. It's more confusing this way when you're creating flags that have nothing to do with surveys

Reply:
Well it would just render double the error message since we're still passing through the validation error from survey flags, which is also not an ideal experience

Reply:
.. No, I mean only one error message, we can choose what to show based on the API response. That's what I meant by looking at the code:

https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/surveys/surveyLogic.tsx#L156

Catch the error, and throw with updated error message if code matches.